### ISSUE TITLE
add a rewind, pause, and forward button for the replays.

### ISSUE THREAD

**Issue by YairMejiaValtierra** [efroemling/ballistica#132]
For better experience and not having to start all over again to see what you want to see

**Comment by Benefit-Zebra** [comment]
Yeah, this will be useful.

**Comment by PythonCode-blip** [comment]
yeah, I was gonna ask the same thing, it's a pretty good suggestion @efroemling since a lot of us have been waiting for this feature to be in-game.

**Comment by Benefit-Zebra** [comment]
But the main problem is that the replay is not in .mp4 but in .brp which is a highly compressed format. So I think it will be hard to add such buttons.

**Comment by PythonCode-blip** [comment]
I think he means adding them in the game for replays instead of the files.

**Comment by Benefit-Zebra** [comment]
I mean that the replay is in .brp format.

**Comment by astrojade** [comment]
Okay, guys, as a heads up, you must be aware that there is already a fast forward button added for bombsquad replays right? 
To access this, just click the menu button , while the replay is going on, and click the minus plus buttons underneath the "end Replay" button. 
![image](https://user-images.githubusercontent.com/67503088/89173123-66d66900-d5a1-11ea-931a-c5dc55197d5f.png)

They are used for speed control.  Incase you have a keyboard attached, you can just click on the Minus and Plus keys to control the speed without having to access the menu section.
I have already spoken to Eric regarding this, a long time ago, technically adding a rewind button was complicated, soo he added only a Speed control button ie from speeds of 0.125 to speeds max upto 8. This makes the replay go in super slow mo if its at speeds of 0.125(min) and super fast if its at speeds of 8. 

There is a reason why the file is saved in a BRP format and not as MP4. These are the reasons:
-> saves your file in the background without adding load on your game
-> reduces storage space 
-> can zoom in and out,  if you have keyboard and mouse attached
-> can adjust the camera in different direction
![image](https://user-images.githubusercontent.com/67503088/89174523-9f774200-d5a3-11ea-823b-e868e6f6af52.png)

if the replay was saved as an mp4, the file size would be huge, you would need a at least minimum 200 GB space to save all the replays. The only way to record your game play is to use a screen recorder or record externally.  Many people don't realize this, but we take such stuff for granted. Many games don't automatically save your replays. You need a good system to handle recording and game-play both at the same time, soo that you get a good Video quality while maintaining fps. But Eric has thought about this, way ahead of time, He gives us the opportunity to auto-save the replay without adding load to the game. Thank You Eric!! Without this feature, i wouldn't be able to record any replay. 

So in order to record your game-play, you need to use a screen recorder to record the game-plays in the replay section. And if your screen recorder adds less load on the system, then it means, better fps, better quality video.

Request to Eric:
Can you please remove the looping for the replay files? The replays always keeps running in a loop, and it doesn't strike me its looping until i see the same scene looping for the 3rd or 4th time.  Can you do something like, the moment the replay reaches the end, it goes back to the replay menu section. By doing this, i don't need to keep staring at the screen, i can do something else and come back later and stop the recording when i see the replay menu section. That way i dont need to keep into account what was the starting scene of the replay and what the end of the replay looks like. (should i edit a video and demonstrate what i am trying to say?)

This will save massive amount of time for me, because i have tons of videos to record, and this would simply make my life much easier. Otherwise what usually happens is that, i screen record the replay, and then go back to the file, and search for the timing where the looping begins, and that is where the hassle begins. What if i closed the screen recorder before i reached the end of the video. where the scene was. I need to redo the screen recording again. But if the video ends with the menu section, then i know that the entire replay got recorded, and thus making editing becomes much easier.

**Comment by ritiek** [comment]
It'd also be nice if we could record chat-messages (along with the gameplay) in .brp replay files.

**Comment by astrojade** [comment]
Yep but i think that it was left out for a reason. Maybe overlays will add extra load or something. No idea. For now i just want the loop removed, because that would be of much better help since we dont have a reverse/back button to go backwards in the replays.
